osu.Game/Beatmaps/Drawables/DifficultyIcon.cs

```diff
@@ -116,6 +117,9 @@ private void load(OsuColour colours)
 
         private Drawable getRulesetIcon()
         {
+            if ((beatmap as APIBeatmap)?.Convert == true)
+                return rulesets.GetRuleset(0)!.CreateInstance().CreateIcon();
```

Comment:
This is pretty ugly; it is not `DifficultyIcon`'s responsibility to be executing this logic this far down. This special casing should be handled in `BeatmapPicker`, or a special constructor overload for `APIBeatmap` specifically that handles the `Convert` property should be provided.